atomate/vasp/firetasks/exchange.py

```diff
@@ -0,0 +1,336 @@
+        cutoff (float): NN search cutoff (Angstrom).
+        tol (float): distance tolerance for similar bonds.
+        average (bool): <J> only or full NN, NNN HeisenbergModel.
+
+    Optional parameters:
+        structures (list): Magnetic structures.
+        energies (list): Energies / atom (eV).
+
+    """
+
+    required_params = [
+        "db_file",
+        "wf_uuid",
+        "cutoff",
+        "tol",
+        "average",
+    ]
+
+    # If only doing <J>, give the original structure/energy inputs
+    optional_params = ["structures", "energies"]
+
+    def run_task(self, fw_spec):
+
+        db_file = env_chk(self["db_file"], fw_spec)
+        wf_uuid = self["wf_uuid"]
+
+        # Get magnetic orderings collection from db
+        mmdb = VaspCalcDb.from_db_file(db_file, admin=True)
+        mmdb.collection = mmdb.db["magnetic_orderings"]
+
+        # Look up static calcs if needed
+        if not self["average"]:
+
+            # Get documents
+            docs = list(
+                mmdb.collection.find(
+                    {"wf_meta.wf_uuid": wf_uuid},
+                    ["task_id", "structure", "energy_per_atom"],
+                )
+            )
+
+            # Get structures and energy / unit cell
+            structures = [Structure.from_dict(d["structure"]) for d in docs]
+            epas = [d["energy_per_atom"] for d in docs]
+            n_atoms = len(structures[0])
+            energies = [e * n_atoms for e in epas]
+        else:
+            structures = self["structures"]
+            energies = self["energies"]
+
+        # Total energies
+        energies = [e * len(s) for e, s in zip(energies, structures)]
+
+        # Map system to a Heisenberg Model
+        hmapper = HeisenbergMapper(structures, energies, self["cutoff"], self["tol"])
+
+        # Get MSONable Heisenberg Model
+        hmodel = hmapper.get_heisenberg_model()
+        hmodel_dict = hmodel.as_dict()
+
+        # Update FW spec with model
+        name = "heisenberg_model_" + str(self["cutoff"])
+        update_spec = {name: hmodel}
+
+        # Write to file
+        dumpfn(hmodel_dict, name + ".json")
+
+        return FWAction(update_spec=update_spec)
+
+
+@explicit_serialize
+class HeisenbergModelToDb(FiretaskBase):
+    """
+    Insert Heisenberg Model object into a DB. Assumes you are in a
+    directory with a model written to a .json file.
+
+    Args:
+        db_file (str): path to file containing the database credentials.
+        wf_uuid (int): Unique id for record keeping.
+        cutoff (float): NN search cutoff (Angstrom).
+
+    """
+
+    required_params = [
+        "db_file",
+        "wf_uuid",
+        "cutoff"
+    ]
+
+    def run_task(self, fw_spec):
+
+        db_file = env_chk(self["db_file"], fw_spec)
+        wf_uuid = self["wf_uuid"]
+
+        name = "heisenberg_model_" + str(self["cutoff"])
+        hmodel = loadfn(name + ".json")
+        #hmodel = HeisenbergModel.from_dict(hmodel_dict)
```

Comment:
The loadfn should be the best way to do this. You can probably remove this commented line.